components/script/dom/windowproxy.rs

```diff
@@ -314,6 +318,26 @@ impl WindowProxy {
         None
     }
 
+    /// https://html.spec.whatwg.org/multipage/#delaying-load-events-mode
+    pub fn is_delaying_load_events_mode(&self) -> bool {
+        self.delaying_load_events_mode.get()
+    }
+
+    /// https://html.spec.whatwg.org/multipage/#delaying-load-events-mode
+    pub fn start_delaying_load_events_mode(&self) {
+        self.delaying_load_events_mode.set(true);
+    }
+
+    /// https://html.spec.whatwg.org/multipage/#delaying-load-events-mode
+    pub fn stop_delaying_load_events_mode(&self) {
+        self.delaying_load_events_mode.set(false);
+        if let Some(document) = self.document() {
+            if !document.loader().events_inhibited() {
```

Comment:
Added this check, otherwise it would fail at https://github.com/servo/servo/blob/f3ca48206e02d48fc8e9f7d340d25e2210d00723/components/script/dom/document.rs#L1907

Reply:
If there is still a loader active, the document will be queued for completion later...